lib/termineter/modules/brute_force_login.py

```diff
@@ -57,61 +58,51 @@ def __iter__(self):
 				yield password
 				password = self.dictionary.readline()
 			self.dictionary.close()
-		raise StopIteration
+		#raise StopIteration; seems to have broken brute forcing multiple user id's
 
 def from_hex(data):
 	return binascii.a2b_hex(data)
 
 def to_hex(data):
 	return binascii.b2a_hex(data).decode('utf-8')
 
+
+
 class Module(TermineterModuleOptical):
 	connection_state = TermineterModuleOptical.connection_states.none
 	def __init__(self, *args, **kwargs):
 		TermineterModuleOptical.__init__(self, *args, **kwargs)
-		self.author = ['Spencer McIntyre']
+		self.author = ['Spencer McIntyre, David Garlak (@vexance)']
 		self.description = 'Brute Force Credentials'
 		self.detailed_description = 'This module is used for brute forcing credentials on the smart meter.  Passwords are not limited to ASCII values and in order to test the entire character space the user will have to provide a dictionary of hex strings and set USE_HEX to true.'
 		self.options.add_boolean('USE_HEX', 'values in word list are in hex', default=True)
 		self.options.add_rfile('DICTIONARY', 'dictionary of passwords to try', required=False, default='$DATA_PATH smeter_passwords.txt')
 		self.options.add_string('USERNAME', 'user name to attempt to log in as', default='0000')
 		self.options.add_integer('USER_ID', 'user id to attempt to log in as', default=1)
+		self.options.add_integer('USER_ID_MAX', 'maximum user id range to attempt bruteforce upon', default=50)
+		self.options.add_boolean('BRUTE_USERS', 'bruteforce user ids from USER_ID to USER_ID_MAX', default=False)
```

Comment:
So from a useability perspective, I'm not a huge fan of options that are only used when other options are set to a particular value. In this case, `USER_ID_MAX` is only used if `BRUTE_USERS` is set.

Instead, a `USER_ID_RANGE` option could be defined that would be greater than or equal to zero and be the number of IDs to bruteforce *after* the existing `USER_ID`. So to bruteforce IDs 2-52, USER_ID would be 2 and USER_ID_RANGE would be 50. Alternatively, you could add a `USER_ID_END` option that when not set would cause the module to just bruteforce the specified `USER_ID` but when set would need to be greater than or equal to the specified `USER_ID`.

Reply:
I liked the idea of the USER_ID_RANGE, and it also simplified things as well. Thanks for the input on this one.